lib/src/segy.c

```diff
@@ -896,8 +896,8 @@ int segy_sample_interval( segy_file* fp, float fallback, float* dt ) {
         return err;
     }
 
-    int bindt = 0;
-    int trdt = 0;
+    int32_t bindt = 0;
+    int32_t trdt = 0;
```

Comment:
Is this because the format is guaranteed to use 32 bit ints?

Reply:
Yes, it's specified as up-to 4-byte two's complement integers. Most are 2-byte ints, but some are 4.